openwisp_controller/tests/test_api.py

```diff
@@ -0,0 +1,39 @@
+# import json
+
+# from django.test import Client, TestCase
+from django.urls import reverse
+from swapper import load_model
+
+from openwisp_controller.config.tests.utils import CreateConfigTemplateMixin
+from openwisp_users.tests.utils import TestMultitenantAdminMixin, TestOrganizationMixin
+
+from .utils import TestAdminMixin
+
+Template = load_model('config', 'Template')
+Vpn = load_model('config', 'Vpn')
+Device = load_model('config', 'Device')
+Config = load_model('config', 'Config')
+
+
+class TestApi(
+    CreateConfigTemplateMixin,
+    TestAdminMixin,
+    TestOrganizationMixin,
+    TestMultitenantAdminMixin,
+):
+    def setUp(self):
+        super().setUp()
+        self._create_org(name='org_a', slug='org_a')
+        self._login()
+
+    def test_get_template_api(self):
+        org1 = self._get_org()
+        print(org1)
+        post_data = self._create_template(organization=org1)
+        print(post_data)
+        response = self.client.post(
+            reverse('controller:template_list'),
+            data=post_data,
+            content_type='application/json',
+        )
+        print(response)
```

Comment:
Only this test for this whole work? This is scary. Any change you introduce we can't be fairly sure that the core feature are working.

This is a red flag. You have to learn to write some tests while you write the main code, this helps you to make sure that the code you're writing works.

Moreover, you have to make sure that the tests you write fail without pieces of code that are added afterwards. That's the only way to make sure that the tests you're writing are correct (because it's totally possible to write bogus tests).

Reply:
:smile: I am working on this.

Reply:
Added the tests. 😀